### ISSUE TITLE
Accessibility: Datagrid on the dialog - context menu pressing TAB key focus goes out side instead of cycle in the menu list.

### ISSUE THREAD

**Issue by sachinkkarande** [vmware/clarity#5926]
## Describe the bug

Datagrid Context/Action menu : Pressing TAB key behaves differently when grid is on dialog and when grid is on application (i.e. not on dialog)

1. When Datagrid is on dialog, TAB key press on the context menu takes focus out of the dialog (Accesibility says focus should remain inside dialog).
2. When Same datagrid is on the application (i.e. not on the dialog), TAB key press cycles the focus inside the context menu

As per Accessibility guidelines focus should not leave the dialog when TAB key is pressed.

*Violation of Accessibility Standards:
WCAG 2.1 -> 2.1.1 Keyboard (Level A)
All functionality of the content is operable through a keyboard interface without requiring specific timings for individual keystrokes, except where the underlying function requires input that depends on the path of the user's movement and not just the endpoints.

## How to reproduce

Check the link: 
https://stackblitz.com/edit/clarity-app-modal-datagrind-context-menu?file=src%2Fapp%2Fmain%2Fmain.component.ts

Here the issue can be seen on the DataGrid present on dialog. DataGrid on the Application works as expected.

Also refer the screen recording "Context Menu focus goes out when on dialog but works fine when on application.mov". 

https://user-images.githubusercontent.com/83701351/117158015-af46df80-addc-11eb-845b-221af4345107.mov

1.  Click button "OPEN MODAL", dialog with Same DataGrid will open
2.  Press TAB twice so that action menu gets selected on he 1st column and 2nd row
3.  Press "ENTER", Action menu/Context Menu will be opened 
4.  Press TAB twice, After "DELETE" menu focus will leave the dialog and some other option outside dialog will get selected.
5. Now, close the dialog (Click Cancel)
6. On the Application, Click some where on the datagrid
7. Press TAB until the action menu square gets selected
8. Now press "ENTER" so that ActionMenu/ContextMenu opens
9. Press TAB repeatedly, observe the behaviour, FOCUS does NOT leave the ContextMenu

## Expected behavior

For the ContextMenu/ActionMenu of DataGrid on the Dialog, after pressing TAB key focus should not go outside the dialog.
Irrespective of the DataGrids location (i.e. on the dialog or not on the dialog) TAB press should work properly and behave SAME on both places.

## Versions

**Clarity project:**

Though the Example version of clarity is "2.4.0", Same behaviour is observed on version "4.0.3".
    "@clr/angular": "4.0.3",
    "@clr/core": "^3.0.0",
    "@clr/icons": "4.0.3",
    "@clr/ui": "4.0.3",

**Device:**

OS: Windows 64bit
Browser: Chrome 87.0.4280.66
VMware: 3.2.0.0.0.17480747
Screen reader: NVDA 2020.3

Note: This defect also occurs on Mac OS with Safari Browser. (Does not seem like a OS/Browser specific issue)

## Additional notes

Previous bug logged for this was https://github.com/vmware/clarity/issues/5065, did not got resolved.

**Comment by gnomeontherun** [comment]
Having our accessibility team review this, but if we are tracking it internally then this is likely a duplicate.

**Comment by scroniser** [comment]
Apologies for the delay: Reviewed and created a new issue to track internally.

**Comment by hippee-lee** [comment]
Thanks @scroniser - I'll close this as it has been triaged and is being tracked with the other a11y issues. 

Clarity cares deeply about accessibility, and we work with our accessibility team to identify and triage issues. We have an internal tracking system to identify and verify accessibility issues, which has captured this already. Some issues with Clarity Angular are difficult to solve without causing side effects or breaking changes, and we have to balance concerns with fixing issues and breaking changes. As we build out Clarity Core, we are able to address these root cause accessibility issues and recommend adopting the Clarity Core implementations as they become available. We are closing this issue as we have the issues tracked internally. Please follow our development and releases to see when we release relevant Clarity Core components to make this possible.
